File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int l = 0;
        int i = 0;
        while(i < s.size() && t.size() ){
            if(s[i] == t[l]){
                l++;
            }
            i++;
        }
        return (t.size() - l);
    }
};

File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        l = 0
        i = 0
        while i < len(s) and l < len(t):
            if s[i] == t[l]:
                l+=1
            i+=1

        return len(t) - l